src/configuration.ts

```diff
@@ -1,24 +1,41 @@
 import { workspace } from 'vscode';
 import * as path from 'path';
 import { existsSync } from 'fs';
+import semver = require('semver');
+import { execSync } from 'child_process';
+import { outputChannel } from './extension';
 
 export interface AmebaConfig {
     command: string;
     configFileName: string;
     onSave: boolean;
+    onType: boolean;
 }
 
 export function getConfig(): AmebaConfig {
     let command = 'ameba';
     const root = workspace.workspaceFolders || [];
     if (root.length) {
         const localAmebaPath = path.join(root[0].uri.fsPath, 'bin', 'ameba');
-        if (existsSync(localAmebaPath)) command = localAmebaPath;
+        if (existsSync(localAmebaPath)) {
+            outputChannel.appendLine(`[Config] Using local ameba at ${localAmebaPath}`)
+            command = localAmebaPath;
+        } else {
+            outputChannel.appendLine(`[Config] Using system ameba`)
+        }
     }
 
+    const workspaceConfig = workspace.getConfiguration('crystal-ameba');
+    const currentVersion = execSync(`"${command}" --version`).toString();
+
+    const onSave = workspaceConfig.get<boolean>("onSave", true);
+    const onType = workspaceConfig.get<boolean>("onType", true) &&
+        semver.satisfies(currentVersion, ">=1.6.2");
```

Comment:
w.r.t. https://github.com/crystal-ameba/ameba/pull/501

```suggestion
        semver.satisfies(currentVersion, ">=1.6.4");
```

Reply:
I don't think this change is necessary as it works currently with `1.6.2`

Reply:
It works in such a way that will lint the whole project when in fact it should lint just one file, wasting a lot of CPU time, and that's unacceptable for me.

Reply:
I re-added passing in `/dev/null`/`nul` locally to get around that.

Reply:
Changed my mind. Thanks for fixing that